### PR TITLE
Changed description of caret character in docs

### LINKED ISSUE #12112: subtle documentation ambiguity
### Problem

The Cargo Book says:

> Caret requirements are an alternative syntax for the default strategy, `^1.2.3` is exactly equivalent to `1.2.3`.

This phrasing has an unfortunate interpretation for Rust newbies, who may take the statement to mean that caret means the exact version identifier.

A very careful reading reveals that caret is merely the default semantic, but is *not* exactly equivalent to any fixed version. Some users may get tripped up by this ambiguity.

### Possible Solution(s)

We can help the reader to disambiguate Cargo syntax and semantics. For example, adjust the section to say something like:

> The caret prefix is also the default version resolution strategy; it may be omitted. Both `^1.2.3` and `1.2.3` match the latest available, *Cargo* SemVer compatible release that Cargo can find.

### PR DESCRIPTION
**Pull request by gautamprikshit1**
Changed the description of caret character in dependency docs. Fixes #12112

### CODE REVIEW

File: src/doc/src/reference/specifying-dependencies.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ using special operators, though it shouldn't be necessary most of the time.
 ### Caret requirements
 
 **Caret requirements** are an alternative syntax for the default strategy,
-`^1.2.3` is exactly equivalent to `1.2.3`.
+`^1.2.3` means the same thing as `1.2.3`, which allows [SemVer]: https://semver.org compatible updates.
```
Comment (weihanglo) [inline src/doc/src/reference/specifying-dependencies.md:59]:
```suggestion
For instance, `log = "^1.2.3"` means the same thing as `log = "1.2.3",
which allows [SemVer] compatible updates.
```

**Review by weihanglo — COMMENTED** [review]
Thanks for being interested in contributing to this project! I guess you're fixing #12112. As a good practice, contributors are encouraged to follow [the process of submitting a PR](https://doc.crates.io/contrib/process/working-on-cargo.html#submitting-a-pull-request) in Cargo Contributor Guide, as well as the guideline in [the pull request template](https://github.com/rust-lang/cargo/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1).

I would suggest providing a minimal description for this pull request.

File: src/doc/src/reference/specifying-dependencies.md (hunk 1 above)
Comment (weihanglo) [inline src/doc/src/reference/specifying-dependencies.md:59]: This combines suggestion from me and ehuss in #12112. What do you think?

```suggestion
For instance, `log = "^1.2.3"` means the same thing as `log = "1.2.3",
which allows [SemVer] compatible updates.
```

File: src/doc/src/reference/specifying-dependencies.md [hunk 2]
```diff
@@ -56,7 +56,8 @@ using special operators, though it shouldn't be necessary most of the time.
 ### Caret requirements
 
 **Caret requirements** are an alternative syntax for the default strategy,
-`^1.2.3` is exactly equivalent to `1.2.3`.
+For instance, `log = "^1.2.3"` means the same thing as `log = "1.2.3"`,
+which allows [SemVer]: https://semver.org compatible updates.
```
Comment (weihanglo) [inline src/doc/src/reference/specifying-dependencies.md:60]: The link is unnecessary I believe?

```suggestion
which allows [SemVer] compatible updates.
```

**Review by weihanglo — CHANGES_REQUESTED** [review]

**Review by weihanglo — APPROVED** [review]
Thanks.

I am going to merge this once CI passes. If you manage to squash commits into one before that, feel free to do so.

**Comment by weihanglo** [comment]
Sorry but no that is not what I meant. I suggested squashing for keeping the git history meaningful. It's not a hard requirement but this PR is better to fit in a single commit.

Some useful resources:

* Here is [an article](https://stackoverflow.com/questions/35703556/what-does-it-mean-to-squash-commits-in-git) about how git squashes works.
* Here is a [CS course from MIT](https://missing.csail.mit.edu/2020/version-control/) that I found it useful to learn version control system.

BTW, communication is essential in open source project collaboration. Feel free to comment here or ask anything.

**Comment by gautamprikshit1** [comment]
I just updated the branch, I will squash the commits soon.
Will join the zulip chat for communication on further issues. I appreciate the resources and feedback.

File: src/doc/src/reference/specifying-dependencies.md [hunk 3]
```diff
@@ -56,7 +56,8 @@ using special operators, though it shouldn't be necessary most of the time.
 ### Caret requirements
 
 **Caret requirements** are an alternative syntax for the default strategy,
```
Comment (LingMan) [inline src/doc/src/reference/specifying-dependencies.md:58]: You're starting a new sentence in the next line, so the comma at the end of this one should be turned into a period.

**Comment by weihanglo** [comment]
@rustbot author

@gautamprikshit1, when this is ready for the next round of review, use `@rustbot ready` to notify reviews.

**Comment by gautamprikshit1** [comment]
@weihanglo couldn't squash commits due to the branch updates between commits and hence large number of commits

**Comment by weihanglo** [comment]
@gautamprikshit1, it seems that you still did it wrong.

d9434e00a62fc145abddc51e86e0cadddd02fa19 is not what LingMan suggests. We probably need to drop that commit. Besides, the commits are still piled up in this pull request. Have you read those aformentioned materials https://github.com/rust-lang/cargo/pull/12209#issuecomment-1571642607? They should provide a decent amount of understanding :)

**Comment by LingMan** [comment]
@gautamprikshit1 My suggestion was to change `, For` to `. For`, not `, which` to `. which`. It's one line earlier.

**Comment by weihanglo** [comment]
Due to inactivity and to reduce the review queue, this will be closed shortly. The patch should be relatively easy. Feel free to submit a new one when the above Git and grammar issues are resolved.

**Pull request closed without merging**